src/hotspot/share/opto/compile.hpp

```diff
@@ -665,7 +667,8 @@ class Compile : public Phase {
   void begin_method();
   void end_method();
   bool should_print_igv(int level);
-  bool should_print_phase(CompilerPhaseType cpt);
+  bool should_print_phase(int level) const;
+  bool should_print_ideal_phase(CompilerPhaseType cpt) const;
```

Comment:
You can use macro `PRODUCT_RETURN_(return false;);` and put both methods under `#ifndef PRODUCT` in .cpp file.

Reply:
Thanks for pointing this out. I actually realized that these functions only have callers from `NOT_PRODUCT` and they implement functionality for development flags. So instead of your suggestion, I moved everything into `#ifndef PRODUCT`.